lib/MiqVm/miq_azure_vm.rb

```diff
@@ -22,9 +22,16 @@ def initialize(azure_handle, args)
       os_disk = vm_obj.properties.storage_profile.os_disk
       if vm_obj.managed_disk?
         #
-        # Use the EVM SNAPSHOT Added by the Provider
+        # Use the Smartstate SNAPSHOT Added by the Provider
+        # If the provider didn't pass the name in then it is running older code that built
+        # the old name - use that instead
         #
-        @disk_name = os_disk.name + "__EVM__SSA__SNAPSHOT"
+        if args[:snapshot]
+          @disk_name = args[:snapshot]
+        else
+          @disk_name = os_disk.name + "__EVM__SSA__SNAPSHOT"
```

Comment:
@jerryk55 Under what conditions will we enter this `else` block? I don't understand when this code will be paired with older provider code.

If we can enter this `else` block, isn't it the same problem we had in the first place?

Reply:
@roliveri I was concerned that there could be a situation where the old provider code was running with the new Smartstate gem and if so then the name of the snapshot would need to be the same as it was previous to this update.  (more on that below)

If we enter this block and the old name is being used, then yes we have an issue if the name is too long - but only if the name is too long - and then we are no worse off than we were before.  

If we aren't passing in the name of the snapshot as an argument then the provider already snapshotted the disk using the old name - if we don't use that name then *any* attempt at running SSA here will fail. If we do use the old name then it will only fail if it is too long.

The above all being said, now that the provider PR has been merged *before* this PR,  this line of execution is actually moot and can probably be removed.  I'll take it out....